# Genres that yield highest net profit

brief description

## import the data

In [1]:
#necessary imports

import sqlite3
import os
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from glob import glob

In [2]:
#changing the display settings to only format floats

pd.options.display.float_format = '{:.2f}'.format

In [3]:
#converts tsv files

tsv_file = './zippedData/rt.movie_info.tsv.gz'
convert = pd.read_table(tsv_file, delimiter='\t')
convert.to_csv('./zippedData/rt.movie_info.csv.gz')

tsv_file_r = './zippedData/rt.reviews.tsv.gz'
convert_r = pd.read_table(tsv_file_r, delimiter='\t', encoding = 'unicode_escape')
convert_r.to_csv('./zippedData/rt.reviews.csv.gz')

In [4]:
#importing data using glob

datafiles = glob("./zippedData/*.csv.gz")
datafiles

['./zippedData\\bom.movie_gross.csv.gz',
 './zippedData\\imdb.name.basics.csv.gz',
 './zippedData\\imdb.title.akas.csv.gz',
 './zippedData\\imdb.title.basics.csv.gz',
 './zippedData\\imdb.title.crew.csv.gz',
 './zippedData\\imdb.title.principals.csv.gz',
 './zippedData\\rt.movie_info.csv.gz',
 './zippedData\\rt.reviews.csv.gz',
 './zippedData\\tmdb.movies.csv.gz',
 './zippedData\\tn.movie_budgets.csv.gz']

In [5]:
datafiles_dict = {}
for filename in datafiles:
    filename_cleaned = os.path.basename(filename).replace(".csv", "").replace(".gz", "").replace(".", "_") # cleaning the filenames
    filename_df = pd.read_csv(filename)
    datafiles_dict[filename_cleaned] = filename_df

In [6]:
print(datafiles_dict.keys())

dict_keys(['bom_movie_gross', 'imdb_name_basics', 'imdb_title_akas', 'imdb_title_basics', 'imdb_title_crew', 'imdb_title_principals', 'rt_movie_info', 'rt_reviews', 'tmdb_movies', 'tn_movie_budgets'])


In [7]:
#load in dataset in csv to dataframe

tmdb_mglb_df = pd.read_csv('tmdb_mglb.csv')

tmdb_mglb_df.head() #call first 5 rows

Unnamed: 0     id                                original_title  \
0           0  12444  Harry Potter and the Deathly Hallows: Part 1   
1           1  10191                      How to Train Your Dragon   
2           2  10138                                    Iron Man 2   
3           3    862                                     Toy Story   
4           4  27205                                     Inception   

   popularity release_date_x                                         title  \
0       33.53     2010-11-19  Harry Potter and the Deathly Hallows: Part 1   
1       28.73     2010-03-26                      How to Train Your Dragon   
2       28.52     2010-05-07                                    Iron Man 2   
3       28.00     1995-11-22                                     Toy Story   
4       27.92     2010-07-16                                     Inception   

   vote_average  vote_count  Fantasy  Drama  ...  Science Fiction  Horror  \
0          7.70       10788     1.00   0.00  ...             0.00    0.00   
1          7.70        7610     1.00   0.00  ...             0.00    0.00   
2          6.80       12368     0.00   0.00  ...             1.00    0.00   
3          7.90       10174     0.00   0.00  ...             0.00    0.00   
4          8.30       22186     0.00   0.00  ...             1.00    0.00   

   original_language  release_date_y                     movie  \
0            English             NaN                       NaN   
1            English    Mar 26, 2010  How to Train Your Dragon   
2            English     May 7, 2010                Iron Man 2   
3            English    Nov 22, 1995                 Toy Story   
4            English    Jul 16, 2010                 Inception   

   production_budget  domestic_gross  worldwide_gross  \
0                nan             nan              nan   
1       165000000.00    217581232.00     494870992.00   
2       170000000.00    312433331.00     621156389.00   
3        30000000.00    191796233.00     364545516.00   
4       160000000.00    292576195.00     835524642.00   

   domesticgross_v_production    total_net  
0                         nan          nan  
1                 52581232.00 329870992.00  
2                142433331.00 451156389.00  
3                161796233.00 334545516.00  
4                132576195.00 675524642.00  

[5 rows x 35 columns]

In [8]:
type(tmdb_mglb_df) #call type of variable

pandas.core.frame.DataFrame

In [9]:
tmdb_mglb_df.shape #return the dimensions of the dataframe array

(26597, 35)

In [10]:
tmdb_mglb_df.info() #return basic summary of dataframe

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 26597 entries, 0 to 26596
Data columns (total 35 columns):
Unnamed: 0                    26597 non-null int64
id                            26597 non-null int64
original_title                26597 non-null object
popularity                    26597 non-null float64
release_date_x                26597 non-null object
title                         26597 non-null object
vote_average                  26597 non-null float64
vote_count                    26597 non-null int64
Fantasy                       26597 non-null float64
Drama                         26597 non-null float64
War                           26597 non-null float64
Mystery                       26597 non-null float64
Comedy                        26597 non-null float64
Romance                       26597 non-null float64
Family                        26597 non-null float64
Music                         26597 non-null float64
Documentary                   26597 non-null float64

In [11]:
tmdb_mglb_df.isna().sum() #sum of all the NaN values in each col

Unnamed: 0                        0
id                                0
original_title                    0
popularity                        0
release_date_x                    0
title                             0
vote_average                      0
vote_count                        0
Fantasy                           0
Drama                             0
War                               0
Mystery                           0
Comedy                            0
Romance                           0
Family                            0
Music                             0
Documentary                       0
Animation                         0
Western                           0
Action                            0
Crime                             0
Thriller                          0
TV Movie                          0
Adventure                         0
History                           0
Science Fiction                   0
Horror                            0
original_language           

In [12]:
tmdb_mglb_df = tmdb_mglb_df.drop(['Unnamed: 0', 'movie', 'title'], axis=1) #drop columns

In [13]:
tmdb_mglb_df.head() #call first 5 rows

id                                original_title  popularity  \
0  12444  Harry Potter and the Deathly Hallows: Part 1       33.53   
1  10191                      How to Train Your Dragon       28.73   
2  10138                                    Iron Man 2       28.52   
3    862                                     Toy Story       28.00   
4  27205                                     Inception       27.92   

  release_date_x  vote_average  vote_count  Fantasy  Drama  War  Mystery  ...  \
0     2010-11-19          7.70       10788     1.00   0.00 0.00     0.00  ...   
1     2010-03-26          7.70        7610     1.00   0.00 0.00     0.00  ...   
2     2010-05-07          6.80       12368     0.00   0.00 0.00     0.00  ...   
3     1995-11-22          7.90       10174     0.00   0.00 0.00     0.00  ...   
4     2010-07-16          8.30       22186     0.00   0.00 0.00     0.00  ...   

   History  Science Fiction  Horror  original_language  release_date_y  \
0     0.00             0.00    0.00            English             NaN   
1     0.00             0.00    0.00            English    Mar 26, 2010   
2     0.00             1.00    0.00            English     May 7, 2010   
3     0.00             0.00    0.00            English    Nov 22, 1995   
4     0.00             1.00    0.00            English    Jul 16, 2010   

   production_budget  domestic_gross  worldwide_gross  \
0                nan             nan              nan   
1       165000000.00    217581232.00     494870992.00   
2       170000000.00    312433331.00     621156389.00   
3        30000000.00    191796233.00     364545516.00   
4       160000000.00    292576195.00     835524642.00   

   domesticgross_v_production    total_net  
0                         nan          nan  
1                 52581232.00 329870992.00  
2                142433331.00 451156389.00  
3                161796233.00 334545516.00  
4                132576195.00 675524642.00  

[5 rows x 32 columns]

## Overall, what are the top 5 grossing films?

In [14]:
#sort values by 4th to last col, descending

sortbytotalgross = tmdb_mglb_df.sort_values(by=['worldwide_gross'], ascending=False)

In [15]:
sortbytotalgross.head()

id          original_title  popularity release_date_x  \
6       19995                  Avatar       26.53     2009-12-18   
23887  299536  Avengers: Infinity War       80.77     2018-04-27   
14236  135397          Jurassic World       20.71     2015-06-12   
14239  168259               Furious 7       20.40     2015-04-03   
5196    24428            The Avengers       50.29     2012-05-04   

       vote_average  vote_count  Fantasy  Drama  War  Mystery  ...  History  \
6              7.40       18676     1.00   0.00 0.00     0.00  ...     0.00   
23887          8.30       13948     1.00   0.00 0.00     0.00  ...     0.00   
14236          6.60       14056     0.00   0.00 0.00     0.00  ...     0.00   
14239          7.30        6538     0.00   0.00 0.00     0.00  ...     0.00   
5196           7.60       19673     0.00   0.00 0.00     0.00  ...     0.00   

       Science Fiction  Horror  original_language  release_date_y  \
6                 1.00    0.00            English    Dec 18, 2009   
23887             0.00    0.00            English    Apr 27, 2018   
14236             1.00    0.00            English    Jun 12, 2015   
14239             0.00    0.00            English     Apr 3, 2015   
5196              1.00    0.00            English     May 4, 2012   

       production_budget  domestic_gross  worldwide_gross  \
6           425000000.00    760507625.00    2776345279.00   
23887       300000000.00    678815482.00    2048134200.00   
14236       215000000.00    652270625.00    1648854864.00   
14239       190000000.00    353007020.00    1518722794.00   
5196        225000000.00    623279547.00    1517935897.00   

       domesticgross_v_production     total_net  
6                    335507625.00 2351345279.00  
23887                378815482.00 1748134200.00  
14236                437270625.00 1433854864.00  
14239                163007020.00 1328722794.00  
5196                 398279547.00 1292935897.00  

[5 rows x 32 columns]

***Findings***: Looking at the total gross revenue of worldwide, including domestic sales, the top 5 grossing films are:
1) Avatar - 2,776,345,279
2) Avengers: Infinity War - 2,048,134,200.00
3) Jurassic World - 1,648,854,864
4) The Avengers - 1,518,722,794
5) Black Panther - 1,517,935,897

In [ ]:
#visual of top 5 grossing

### If available, what are the genres of those top 5 grossing films? 

In [16]:
code here: KAT

SyntaxError: invalid syntax (<ipython-input-16-d4d7f2c1cec0>, line 1)

***Findings***: Leveraged the [imdb_title_basics] dataset to find the titles from top 5 grossing films and noted identified genres for each film

## Top 5 grossing films by year

### What are the titles of the films?

In [ ]:
conn = sqlite3.connect('movie.db') 
cur = conn.cursor()

cur.execute(""" 
SELECT 
  DISTINCT year, domestic_gross, movie
FROM tn_movie_budgets 
ORDER BY domestic_gross DESC
;
""")

dft = pd.DataFrame(cur.fetchall())
dft.columns = (x[0] for x in cur.description)
#print(dft.shape)
#print(dft.info())

dft.head()

### What was the production cost?

In [ ]:
conn = sqlite3.connect('movie.db') 
cur = conn.cursor()

cur.execute(""" 
SELECT 
  DISTINCT year, production_budget, movie
FROM tn_movie_budgets 
ORDER BY domestic_gross DESC
;
""")

dft = pd.DataFrame(cur.fetchall())
dft.columns = (x[0] for x in cur.description)
#print(dft.shape)
#print(dft.info())

dft.head()

***Findings***: write summary here

### What was the runtime?

In [ ]:
code here: KAT

***Findings***: write summary here >> Leveraged the [imdb_title_basics] dataset to find the titles from top 5 grossing films and noted the runtime for each film

## Most recently, what has been the average production cost? 

In [22]:
#sort by production budget

productioncosts = tmdb_mglb_df.sort_values(by=['production_budget'], ascending=False)

productioncosts.head()

id                               original_title  popularity  \
6       19995                                       Avatar       26.53   
2481     1865  Pirates of the Caribbean: On Stranger Tides       30.58   
14211   99861                      Avengers: Age of Ultron       44.38   
14217  206647                                      Spectre       30.32   
17440  206647                                      Spectre       30.32   

      release_date_x  vote_average  vote_count  Fantasy  Drama  War  Mystery  \
6         2009-12-18          7.40       18676     1.00   0.00 0.00     0.00   
2481      2011-05-20          6.40        8571     1.00   0.00 0.00     0.00   
14211     2015-05-01          7.30       13457     0.00   0.00 0.00     0.00   
14217     2015-11-06          6.40        6719     0.00   0.00 0.00     0.00   
17440     2015-11-06          6.40        6719     0.00   0.00 0.00     0.00   

       ...  History  Science Fiction  Horror  original_language  \
6      ...     0.00             1.00    0.00            English   
2481   ...     0.00             0.00    0.00            English   
14211  ...     0.00             1.00    0.00            English   
14217  ...     0.00             0.00    0.00            English   
17440  ...     0.00             0.00    0.00            English   

       release_date_y  production_budget  domestic_gross  worldwide_gross  \
6        Dec 18, 2009       425000000.00    760507625.00    2776345279.00   
2481     May 20, 2011       410600000.00    241063875.00    1045663875.00   
14211     May 1, 2015       330600000.00    459005868.00    1403013963.00   
14217     Nov 6, 2015       300000000.00    200074175.00     879620923.00   
17440     Nov 6, 2015       300000000.00    200074175.00     879620923.00   

       domesticgross_v_production     total_net  
6                    335507625.00 2351345279.00  
2481                -169536125.00  635063875.00  
14211                128405868.00 1072413963.00  
14217                -99925825.00  579620923.00  
17440                -99925825.00  579620923.00  

[5 rows x 32 columns]

In [23]:
#remove rows with missing values in place/within the original dataframe

productioncosts.dropna(inplace = True) 

productioncosts

id                               original_title  popularity  \
6       19995                                       Avatar       26.53   
2481     1865  Pirates of the Caribbean: On Stranger Tides       30.58   
14211   99861                      Avengers: Age of Ultron       44.38   
14217  206647                                      Spectre       30.32   
17440  206647                                      Spectre       30.32   
...       ...                                          ...         ...   
12365  279516                                          Ten        1.57   
14730  226458                                       Exeter        5.93   
10500  255266                                    Dry Spell        0.60   
8920    86304                     All Superheroes Must Die        2.08   
3484    72766                                    Newlyweds        1.97   

      release_date_x  vote_average  vote_count  Fantasy  Drama  War  Mystery  \
6         2009-12-18          7.40       18676     1.00   0.00 0.00     0.00   
2481      2011-05-20          6.40        8571     1.00   0.00 0.00     0.00   
14211     2015-05-01          7.30       13457     0.00   0.00 0.00     0.00   
14217     2015-11-06          6.40        6719     0.00   0.00 0.00     0.00   
17440     2015-11-06          6.40        6719     0.00   0.00 0.00     0.00   
...              ...           ...         ...      ...    ...  ...      ...   
12365     2014-03-28          5.40           5     0.00   0.00 0.00     1.00   
14730     2015-03-26          4.70         121     0.00   0.00 0.00     0.00   
10500     2013-02-14          6.00           1     0.00   0.00 0.00     0.00   
8920      2013-01-04          3.90          19     0.00   0.00 0.00     0.00   
3484      2011-12-26          5.40           7     0.00   0.00 0.00     0.00   

       ...  History  Science Fiction  Horror  original_language  \
6      ...     0.00             1.00    0.00            English   
2481   ...     0.00             0.00    0.00            English   
14211  ...     0.00             1.00    0.00            English   
14217  ...     0.00             0.00    0.00            English   
17440  ...     0.00             0.00    0.00            English   
...    ...      ...              ...     ...                ...   
12365  ...     0.00             0.00    1.00            English   
14730  ...     0.00             0.00    1.00            English   
10500  ...     0.00             0.00    0.00            English   
8920   ...     0.00             1.00    0.00            English   
3484   ...     0.00             0.00    0.00            English   

       release_date_y  production_budget  domestic_gross  worldwide_gross  \
6        Dec 18, 2009       425000000.00    760507625.00    2776345279.00   
2481     May 20, 2011       410600000.00    241063875.00    1045663875.00   
14211     May 1, 2015       330600000.00    459005868.00    1403013963.00   
14217     Nov 6, 2015       300000000.00    200074175.00     879620923.00   
17440     Nov 6, 2015       300000000.00    200074175.00     879620923.00   
...               ...                ...             ...              ...   
12365    Apr 21, 2015           25000.00            0.00             0.00   
14730     Sep 1, 2015           25000.00            0.00        489792.00   
10500    Dec 31, 2014           22000.00            0.00             0.00   
8920      Jan 4, 2013           20000.00            0.00             0.00   
3484     Jan 13, 2012            9000.00         4584.00          4584.00   

       domesticgross_v_production     total_net  
6                    335507625.00 2351345279.00  
2481                -169536125.00  635063875.00  
14211                128405868.00 1072413963.00  
14217                -99925825.00  579620923.00  
17440                -99925825.00  579620923.00  
...                           ...           ...  
12365                   -25000.00     -25000.00  
14730                   -25000.

In [26]:
#calculate median of production_budget column

productioncosts.loc[:, 'production_budget'].median()

19000000.0

***Findings***: Median/Average production costs - 19,000,000

## Which 5 genres had the highest box office revenue?

In [ ]:
code here: KAT

***Findings***: In the Rotten Tomatoes dataset, Is there a common genre associated with highest box office revenue?

## Data Visualization

In [ ]:
code here

***Findings***: write summary here